py/examples/tour.py

```diff
@@ -29,24 +29,45 @@ def __init__(self, filename: str, title: str, description: str, source: str):
         self.code = highlight(source, py_lexer, html_formatter)
         self.previous_example: Optional[Example] = None
         self.next_example: Optional[Example] = None
-        self.process: Optional[asyncio.subprocess.Process] = None
+        self.process: Optional[subprocess.Popen] = None
         self.is_app = source.find('@app(') > 0
 
     async def start(self):
         if self.is_app:
-            self.process = await asyncio.create_subprocess_exec(
-                sys.executable, '-m', 'uvicorn', '--port', _app_port, f'examples.{self.name}:main', env=dict(
-                    H2O_WAVE_EXTERNAL_ADDRESS=f'http://{_app_host}:{_app_port}'
-                ),
-            )
+            self.process = subprocess.Popen(
+                [sys.executable, '-m', 'uvicorn', '--port', _app_port, f'examples.{self.name}:main'],
+                env=self.adjust_env_for_platform(dict(H2O_WAVE_EXTERNAL_ADDRESS=f'http://{_app_host}:{_app_port}')))
         else:
-            self.process = await asyncio.create_subprocess_exec(
-                sys.executable, os.path.join(example_dir, self.filename)
-            )
+            self.process = subprocess.Popen([sys.executable, os.path.join(example_dir, self.filename)])
 
     async def stop(self):
         if self.process and self.process.returncode is None:
             self.process.terminate()
+            self.process.wait()
```

Comment:
Is this `wait()` necessary?

Reply:
If we don't call wait(). Zombie process entries remain in the process table until the user terminates the parent process in MasOS.
<img width="1785" alt="Screenshot 2020-12-22 at 08 11 59" src="https://user-images.githubusercontent.com/5285626/102842785-cb4d7400-442d-11eb-838d-ce23f6ee8e31.png">